src/Splitio/Services/Evaluator/Evaluator.cs

```diff
@@ -70,27 +71,67 @@ public MultipleEvaluatorResult EvaluateFeatures(Key key, List<string> featureNam
                 }
                 catch (Exception e)
                 {
-                    _log.Error($"Exception caught getting treatments", e);
+                    exception = EvaluateFeaturesException(e, featureNames, clock, out treatmentsForFeatures);
+                }
 
-                    foreach (var name in featureNames)
-                    {
-                        treatmentsForFeatures.Add(name, new TreatmentResult(Labels.Exception, Control, elapsedMilliseconds: clock.ElapsedMilliseconds));
-                    }
+                return new MultipleEvaluatorResult(treatmentsForFeatures, clock.ElapsedMilliseconds, exception);
+            }
+        }
+        #endregion
 
-                    exception = true;
+        #region Public Async Methods
+        public async Task<TreatmentResult> EvaluateFeatureAsync(Key key, string featureName, Dictionary<string, object> attributes = null)
+        {
+            using (var clock = new SplitStopwatch())
+            {
+                clock.Start();
+
+                try
+                {
+                    var parsedSplit = await _featureFlagCacheConsumer.GetSplitAsync(featureName);
+
+                    return await EvaluateTreatmentAsync(key, parsedSplit, featureName, clock, attributes);
                 }
+                catch (Exception e)
+                {
+                    return EvaluateFeatureException(e, featureName, clock);
```

Comment:
Is it ok, or should it be something like `return await EvaluateFeatureExceptionAsync(...)`?